src/Microsoft.Azure.WebJobs.ServiceBus/MessagingProvider.cs

```diff
@@ -17,11 +17,7 @@ namespace Microsoft.Azure.WebJobs.ServiceBus
     public class MessagingProvider
     {
         private readonly ServiceBusConfiguration _config;
-
-        /// <summary>
-        /// Cache of <see cref="MessagingFactory"/> instances by connection string.
-        /// </summary>
-        private readonly ConcurrentDictionary<string, MessagingFactory> _messagingFactoryCache = new ConcurrentDictionary<string, MessagingFactory>();
+        private readonly ConcurrentDictionary<Tuple<string, string>, MessageReceiver> _messageReceiverCache = new ConcurrentDictionary<Tuple<string, string>, MessageReceiver>();
```

Comment:
Previously we cached the `MessagingFactory`, but you still got a new `MessageReceiver` every time you called `CreateMessageReceiver` -- at least that's what it looks like. I can't tell if the `MessagingFactory` cached them internally or not. 

This change means that you may get the same `MessageReceiver` if you call `CreateMessageReceiver` twice with the same details.

Does that matter? It may not -- having two receivers watching the same queue would flip between them anyway.

Reply:
Is this not relevant?